### ISSUE TITLE
OpenShift deployment fails initializing container

### ISSUE THREAD

**Issue by ctron** [eclipse/hono#572]
Coming from #566 it looks to me like the openshift deployments (both EnMasse and "plain") currently have an issue starting up properly due to an initialization error of the device registry init container:

~~~
➜  openshift git:(master) ✗ oc logs hono-service-device-registry-1-7bttt -c copy-example-data
cp: can't create '/var/lib/hono/device-registry/credentials.json': Permission denied
cp: can't create '/var/lib/hono/device-registry/tenants.json': Permission denied
~~~

Currently I am not sure what is wrong as the volume should have the proper permissions.

**Comment by ctron** [comment]
Ok, so I guess the whole problem is the PV/PVC for the device registry. Minishift provides support for automatically providing PVs for application PVCs. However Hono currently has some extra selector in the PVC which prevents the PVs to bind:

~~~yml
  selector:
    matchLabels:
      group: ${project.groupId}
      project: hono-example
~~~

@dejanb @sophokles73 is there any reason for this specific selector? Dropping it solves the issue for me.

**Comment by ctron** [comment]
I think I can patch this out during the openshift setup. But I would prefer to remove it altogether if there is no specific reason for this anyway.

**Comment by sophokles73** [comment]
> is there any reason for this specific selector? Dropping it solves the issue for me.

Well, the reason for it was that minishift did *not* support automatic creation of the PV in the past. This is also the reason why we needed to introduce the manual step of creating the PV folder in the filesystem on the minishift VM as part of the deployment instructions ...

**Comment by sophokles73** [comment]
Are you creating the filesystem folder on the minishift VM as described in https://www.eclipse.org/hono/deployment/openshift/ before you deploy?

I am deploying master to Minishift 1.15.1 with OpenShift 3.7.2 following the instructions with no problems ... well, there is a problem deploying influxdb and grafana because openshift does not support the apps/v1beta2 Deployment resource but that is another problem.

**Comment by sophokles73** [comment]
I have just pushed the change to use apps/v1beta1 Deployment for artemis and dispatch router (ec6c19787b87ba699692b59a5965d6cd116a38d0)

**Comment by ctron** [comment]
I did create the folder according to spec. Still it didn't work. However I think we can drop everything now. The selector, the additional step, ... WDYT?

**Comment by ctron** [comment]
I was using 1.13 and now 1.15.

**Comment by ctron** [comment]
I do think this is closed by #575
